### PR TITLE
manually-merge: cache data-sources in CollectionNameResolver

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

File: arangod/Utils/CollectionNameResolver.cpp [hunk 1]
```diff
@@ -32,7 +32,33 @@
...
 #include "VocBase/vocbase.h"
 
-using namespace arangodb;
+namespace arangodb {
+
+std::shared_ptr<LogicalCollection> CollectionNameResolver::getCollection(
+  TRI_voc_cid_t id
+) const noexcept {
+  #ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+    return std::dynamic_pointer_cast<LogicalCollection>(getDataSource(id));
```
Comment (gnusi) [inline arangod/Utils/CollectionNameResolver.cpp]: please add TRI_ASSERT:
```
auto col = std::dynamic_pointer_cast<LogicalCollection>(getDataSource(id))
TRI_ASSERT(col)
return col
```

File: arangod/Utils/CollectionNameResolver.cpp [hunk 2]
```diff
@@ -32,7 +32,33 @@
...
+  return dataSource.category() == LogicalCollection::category()
+    ? std::static_pointer_cast<LogicalCollection>(dataSource) : nullptr;
+  #endif
+}
+
+std::shared_ptr<LogicalCollection> CollectionNameResolver::getCollection(
+  std::string const& nameOrId
+) const noexcept {
+  #ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+    return std::dynamic_pointer_cast<LogicalCollection>(getDataSource(nameOrId));
```
Comment (gnusi) [inline arangod/Utils/CollectionNameResolver.cpp]: please add TRI_ASSERT:
```
auto col = std::dynamic_pointer_cast<LogicalCollection>(getDataSource(id))
TRI_ASSERT(col)
return col
```

File: arangod/Utils/CollectionNameResolver.cpp [hunk 3]
```diff
@@ -319,36 +303,101 @@ std::string CollectionNameResolver::localNameLookup(TRI_voc_cid_t cid) const {
...
 
-  return nullptr;
+  return ptr;
+}
+
+std::shared_ptr<LogicalView> CollectionNameResolver::getView(
+  TRI_voc_cid_t id
+) const noexcept {
+  #ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+    return std::dynamic_pointer_cast<LogicalView>(getDataSource(id));
```
Comment (gnusi) [inline arangod/Utils/CollectionNameResolver.cpp]: please add TRI_ASSERT:
```
auto view = std::dynamic_pointer_cast<LogicalView>(getDataSource(id))
TRI_ASSERT(view)
return view
```

File: arangod/Utils/CollectionNameResolver.cpp [hunk 4]
```diff
@@ -319,36 +303,101 @@ std::string CollectionNameResolver::localNameLookup(TRI_voc_cid_t cid) const {
...
+  return dataSource.category() == LogicalView::category()
+    ? std::static_pointer_cast<LogicalView>(dataSource) : nullptr;
+  #endif
+}
+
+std::shared_ptr<LogicalView> CollectionNameResolver::getView(
+  std::string const& nameOrId
+) const noexcept {
+  #ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+    return std::dynamic_pointer_cast<LogicalView>(getDataSource(nameOrId));
```
Comment (gnusi) [inline arangod/Utils/CollectionNameResolver.cpp]: please add TRI_ASSERT:
```
auto view = std::dynamic_pointer_cast<LogicalView>(getDataSource(id))
TRI_ASSERT(view)
return view
```

File: arangod/Utils/CollectionNameResolver.cpp (hunk 1 above)
Comment (vasiliy-arangodb) [reply arangod/Utils/CollectionNameResolver.cpp]: no need for TRI_ASSERT, it is valid to return nullptr

File: arangod/Utils/CollectionNameResolver.cpp (hunk 2 above)
Comment (vasiliy-arangodb) [reply arangod/Utils/CollectionNameResolver.cpp]: no need for TRI_ASSERT, it is valid to return nullptr

File: arangod/Utils/CollectionNameResolver.cpp (hunk 3 above)
Comment (vasiliy-arangodb) [reply arangod/Utils/CollectionNameResolver.cpp]: no need for TRI_ASSERT, it is valid to return nullptr

File: arangod/Utils/CollectionNameResolver.cpp (hunk 4 above)
Comment (vasiliy-arangodb) [reply arangod/Utils/CollectionNameResolver.cpp]: no need for TRI_ASSERT, it is valid to return nullptr

**Comment by vasiliy-arangodb** [comment]
abandoned in favour of https://github.com/arangodb/arangodb/pull/4837 as per review request

**Pull request closed without merging**